test/common/http/conn_manager_impl_test.cc

```diff
@@ -2700,6 +2700,40 @@ TEST_F(HttpConnectionManagerImplTest, FilterHeadReply) {
   conn_manager_->onData(fake_input, false);
 }
 
+TEST_F(HttpConnectionManagerImplTest, FilterHeaderOnly) {
```

Comment:
Please make sure you have test coverage for all permutations:
1) Header only request/response
2) Request/response ended with data frame
4) Request/response ended with trailers